diff --git a/dist/index.d.ts b/dist/index.d.ts
index 06d01d4005d2174c383e0561ba971540899ded3e..be3ee867c06c326ec56cdedff5dc2531fb99906b 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -100,17 +100,14 @@ export type RequestBodyOption<T> = OperationRequestBodyContent<T> extends never
 
 export type FetchOptions<T> = RequestOptions<T> & Omit<RequestInit, "body" | "headers">;
 
-export type FetchResponse<T extends Record<string | number, any>, Options, Media extends MediaType> =
-  | {
-      data: ParseAsResponse<SuccessResponse<ResponseObjectMap<T>, Media>, Options>;
-      error?: never;
-      response: Response;
-    }
-  | {
-      data?: never;
-      error: ErrorResponse<ResponseObjectMap<T>, Media>;
-      response: Response;
-    };
+export type FetchResponse<
+  T extends Record<string | number, any>,
+  Options,
+  Media extends MediaType,
+> = {
+  data: ParseAsResponse<SuccessResponse<ResponseObjectMap<T>, Media>, Options>;
+  response: Response;
+};
 
 export type RequestOptions<T> = ParamsOption<T> &
   RequestBodyOption<T> & {
